keras_cv/layers/preprocessing3d/frustum_random_dropping_points.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv.layers.preprocessing3d import base_augmentation_layer_3d
+from keras_cv.ops import point_cloud
+
+POINT_CLOUDS = base_augmentation_layer_3d.POINT_CLOUDS
+BOUNDING_BOXES = base_augmentation_layer_3d.BOUNDING_BOXES
+
+
+class FrustumRandomDroppingPoints(base_augmentation_layer_3d.BaseAugmentationLayer3D):
+    """A preprocessing layer which randomly drops point within a randomly generated frustum during training.
+
+    This layer will randomly select a point from the point cloud as the center of a frustum then generate a frustum based
+    on r_distance, theta_width, and phi_width. Points inside the selected frustum are randomly dropped based on keep_probability.
```

Comment:
nit: based on drop_rate? (since that's the param name)

Reply:
Done.